core/src/main/scala/kafka/utils/Time.scala

```diff
@@ -44,6 +46,8 @@ trait Time {
 
   def nanoseconds: Long
 
+  def hiResClockMs: Long = TimeUnit.NANOSECONDS.toMillis(nanoseconds)
```

Comment:
OK, I understand now, my bad. You are relying on `Time.nanoseconds` in your implementation instead of `System.nanoTime`. Thanks for explaining. :)